storage/innobase/trx/trx0trx.cc

```diff
@@ -1534,6 +1536,41 @@ trx_commit_in_memory(
 	}
 }
 
+/*************************************************************//**
+Update the persistent counts of all tables modified by a transaction. */
+void trx_update_persistent_counts(
+/*============*/
+    trx_t* trx)		/*!< in: transaction */
+{
+    dict_table_t* ib_table;
+    TABLE* table;
+    bool update_persistent_count;
+    for (trx_mod_tables_t::const_iterator t = trx->mod_tables.begin();
+         t != trx->mod_tables.end(); t++) {
+        ib_table = t->first;
+
+        table = thd_get_open_tables(current_thd);
+        for (; table; table = table->next) {
+            if (ib_table == ((ha_innobase *) table->file)->m_prebuilt->table) {
+                break;
+            }
+        }
+
+        if (table) {
```

Comment:
Can we avoid the search loop above, and simply refer to `ib_table->m_prebuilt->table`?

Reply:
I'm trying to map an Innobase `dict_sys_table` to a SQL `TABLE`. Not sure I follow what you mean by 
> refer to `ib_table->m_prebuilt->table`?

Unless you mean iterating through `thd->open_tables` instead of `trx->mod_tables` in the outer loop as in:
```
for (TABLE* table = trx->mysql_thd->open_tables; table; table = table->next) {
    ib_table = ((ha_innobase *) table->file)->m_prebuilt->table;
    ...
    innobase_update_persistent_count(ib_table, table, trx);
    ...
}
```

Reply:
I don’t think that we need a `TABLE` at all. I think that we must strip down `innobase_update_persistent_count()` to the bare minimum, so that it will directly update the metadata BLOB page in a mini-transaction. We could cache the metadata BLOB page number in `dict_table_t`, so that there is no need to find the metadata record and follow the pointer there first.